dialer.go

```diff
@@ -297,24 +301,33 @@ func (d *Dialer) Close() error {
 	return nil
 }
 
-func (d *Dialer) instance(connName string) (*cloudsql.Instance, error) {
+// instance is a helper function for returning the appropriate instance object in a threadsafe way.
+// It will create a new instance object, modify the existing one, or leave it unchanged as needed.
+func (d *Dialer) instance(connName string, rCfg *cloudsql.RefreshCfg) (*cloudsql.Instance, error) {
 	// Check instance cache
 	d.lock.RLock()
 	i, ok := d.instances[connName]
 	d.lock.RUnlock()
-	if !ok {
+	// Check if the instance exists and that the refresh cfg is the same
+	if !ok || (rCfg != nil && *rCfg != i.RefreshCfg) {
```

Comment:
nit: `(rCfg != nil && *rCfg != i.RefreshCfg)` might be clearer as a function declared inline.

``` golang
newConfig := func(r *cloudsql.RefreshCfg) bool { return r != nil && *r != i.RefreshCfg }
if !ok || newConfig(rCfg) { /* ... */ }
```

Reply:
Adding a function seems to obfuscate for me. I updated the comment here instead.